falcon/request.py

```diff
@@ -256,6 +257,9 @@ def __init__(self, env, options=None):
         self.stream = env['wsgi.input']
         self.method = env['REQUEST_METHOD']
 
+        if self.stream.__class__.__name__ == InputWrapper.__name__:
```

Comment:
Alternatively, would adding `InputWrapper` to the `isinstance()` check on [L306](https://github.com/falconry/falcon/blob/master/falcon/request.py#L306) do the trick?